### ISSUE TITLE
Change jMock to Mockito

### ISSUE THREAD

**Issue by cxxr** [cxxr/better-java#50]
Also mention EasyMock as another alternative. Idea from PR #49.

**Comment by erosb** [comment]
Yup, I've checked Mockito too (due to the PR #49  ) but it seems mockito 1.x is obsolete already, while 2.0 is still in beta. I personally don't recommend changing to any of these at the moment (lets wait a few months with this change).

**Comment by vincecima** [comment]
@erosb what about 1.x is obsolete?

**Comment by erosb** [comment]
I meant that if 2.x is in beta then probably they will roll out a stable release soon, and in that moment 1.x will be the "old" mockito. I do not recommend to include in the better-java README a version which soon will be old. I would also discourage recommending the 2.x library since it is - at the moment - in beta.

Of course I do not think that anyone using mockito 1.x has to stop using it and move to 2.x, so my above comment was not meant to be offensive, feel free to keep using 1.x if that is what you are familiar with. But it makes a difference what to recommend to people unfamiliar with mocking at all, and what should be used by experienced testers.

To sum up I recommend staying with the jMock example (since it turned out to be quite popular in #49 ) and lets reconsider Mockito when 2.x is out.

**Comment by smoyer64** [comment]
As I mentioned in #49, I think better-java should describe mocking as a
best practice but wouldn't be opposed to having examples using more than
one framework.  For those of us using CDI, the CDI-Unit library also
includes the ability to Mock objects annotated with @Inject.

On Mon, Dec 7, 2015 at 12:57 PM, Bence Eros <notifications@github.com>
wrote:

> I meant that if 2.x is in beta then probably they will roll out a stable
> release soon, and in that moment 1.x will be the "old" mockito. I do not
> recommend to include in the better-java README a version which soon will be
> old. I would also discourage recommending the 2.x library since it is - at
> the moment - in beta.
>
> Of course I do not think that anyone using mockito 1.x has to stop using
> it and move to 2.x, so my above comment was not meant to be offensive, feel
> free to keep using 1.x if that is what you are familiar with. But it makes
> a difference what to recommend to people unfamiliar with mocking at all,
> and what should be used by experienced testers.
>
> To sum up I recommend staying with the jMock example (since it turned out
> to be quite popular in #49 <https://github.com/cxxr/better-java/pull/49>
> ) and lets reconsider Mockito when 2.x is out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cxxr/better-java/issues/50#issuecomment-162607791>.
>



-- 
Sent from my iBerry MacPhone
